cdci_data_analysis/analysis/parameters.py

```diff
@@ -584,78 +555,74 @@ def _split(self,str_list):
 
     @property
     def value(self):
-        if self._value==[''] or self._value is None:
+        if self._value == [''] or self._value is None:
             return []
         else:
             return self._value
 
     @value.setter
     def value(self, v):
-        #print('set', self.name, v, self._allowed_values)
+        # print('set', self.name, v, self._allowed_values)
         if v is not None:
             if self.check_value is not None:
                 self.check_value(v, units=self.units, name=self.name)
             if self._allowed_values is not None:
                 if v not in self._allowed_values:
                     raise RuntimeError('value', v, 'not allowed, allowed=', self._allowed_values)
             if v == [''] or v is None or str(v) == '':
-                self._value=['']
+                self._value = ['']
             else:
                 self._value = v
         else:
             self._value = ['']
-        self._value=self._split(self._value)
-        #print ('set to ',self._value)
-
+        self._value = self._split(self._value)
+        # print ('set to ',self._value)
 
     @staticmethod
-    def check_list_value(value,units,name='par'):
-        if units=='names_list':
-            try:
-                #print(type(value))
-                assert (type(value) == list or type(value) == str  or type(str(value))== str)
-            except:
-                raise RuntimeError('par:',name,', value is not product list format : list of strings','it is',type(value),value)
+    def check_list_value(value, units, name='par'):
+        if units == 'names_list':
+            # TODO the condition 'isinstance(str(value), str))' was quite unclear to me, and probably useless since could lead to unexpected behavior
+            if not (isinstance(value, list) or isinstance(value, str)
+                    or isinstance(value, float) or isinstance(value, int)):
+                raise RuntimeError(f'value of the parameter {name} is not a valid product list format, but {type(value)} has been found')
         else:
-            raise  RuntimeError(name,'units not valid',units)
-
-
-
-
+            raise RuntimeError(name, 'units not valid', units)
 
 
 class Angle(Parameter):
-        def __init__(self,value=None, units=None,name=None):
-
-            super(Angle, self).__init__(value=value,
-                                       units=units,
-                                       name=name,
-                                       allowed_units=None)
-            # wtform_dict=wtform_dict)
-
-
-            self._set_angle(value, units=units)
+    def __init__(self, value=None, units=None, name=None):
+
+        super().__init__(value=value,
+                         units=units,
+                         # TODO can we safely make this assumption?
+                         default_units='deg',
```

Comment:
we can make it in principle, but we should not really add `default_units` without `units_name`.

Reply:
good point, let's not add it for now then

Reply:
If possible, there should be `units_name` for Energy. Something like `E1_units`. Actually, it might be defined in the plugin, depending on the parameter name.
Maybe then no change is needed here.